mlflow/store/artifact/cloud_artifact_repo.py

```diff
@@ -0,0 +1,270 @@
+                    inflight_uploads[staged_upload.src_file_path] = upload_future
+
+                yield from inflight_uploads.items()
+
+        with ArtifactProgressBar.files(
+            desc="Uploading artifacts", total=len(staged_uploads)
+        ) as pbar:
+            if len(staged_uploads) >= 10 and pbar.pbar:
+                _logger.info(
+                    "The progress bar can be disabled by setting the environment "
+                    f"variable {MLFLOW_ENABLE_ARTIFACTS_PROGRESS_BAR} to false"
+                )
+            for src_file_path, upload_future in upload_artifacts_iter():
+                try:
+                    upload_future.result()
+                    pbar.update()
+                except Exception as e:
+                    failed_uploads[src_file_path] = repr(e)
+
+        if len(failed_uploads) > 0:
+            raise MlflowException(
+                message=(
+                    "The following failures occurred while uploading one or more artifacts"
+                    f" to {self.artifact_uri}: {failed_uploads}"
+                )
+            )
+
+    @abstractmethod
+    def _get_write_credential_infos(self, remote_file_paths):
+        """
+        Retrieve write credentials for a batch of remote file paths, including presigned URLs.
+
+        :param remote_file_paths: List of destination file paths on the remote server.
+        :return: List of ArtifactCredentialInfo objects corresponding to each file path.
+        """
+        pass
+
+    @abstractmethod
+    def _upload_to_cloud(self, cloud_credential_info, src_file_path, artifact_file_path):
+        """
+        Upload a single file to the cloud.
+
+        :param cloud_credential_info: ArtifactCredentialInfo object with presigned URL for the file.
+        :param src_file_path: Local source file path for the upload.
+        :param artifact_file_path: Path in the artifact repository where the artifact will be
+                                   logged.
+        :return:
+        """
+        pass
+
+    # Read APIs
+
+    def _extract_headers_from_credentials(self, headers):
+        """
+        :return: A python dictionary of http headers converted from the protobuf credentials
+        """
+        return {header.name: header.value for header in headers}
+
+    def _parallelized_download_from_cloud(self, file_size, remote_file_path, local_path):
+        read_credentials = self._get_read_credential_infos([remote_file_path])
+        # Read credentials for only one file were requested. So we expected only one value in
+        # the response.
+        assert len(read_credentials) == 1
+        cloud_credential_info = read_credentials[0]
+        # from mlflow.utils.databricks_utils import get_databricks_env_vars
+
+        try:
+            parallel_download_subproc_env = os.environ.copy()
+            # parallel_download_subproc_env.update(
+            #     get_databricks_env_vars(self.databricks_profile_uri)
+            # )
```

Comment:
Still investigating whether this is necessary. It could be tricky to add a `databricks_profile_uri` to every subclass.

Reply:
@harupy everything seemed to worked without this, whether in databricks or locally (auth configured via Databricks CLI), for all three clouds (see updated PR description). Am I missing a case that I haven't tested?

Reply:
`get_databricks_env_vars` has this comment:

```
    # We set these via environment variables so that only the current profile is exposed, rather
    # than all profiles in ~/.databrickscfg; maybe better would be to mount the necessary
    # part of ~/.databrickscfg into the container
```

@dbczumar Any risk if we use all profiles in ~/.databrickscfg?

Reply:
If the user does `os.environ["DATABRICKS_TOKEN"] = ...` / `os.environ["DATABRICKS_HOST"] = "..."`, none of those settings will be propagated to the subprocess, causing errors. I think we have to propagate the env vars

Reply:
That makes sense. Let's uncomment this line.

Reply:
Update: we discussed offline that we won't uncomment this line because we have `parallel_download_subproc_env = os.environ.copy()` in this PR and manually testing with environment variable overrides worked.